### PR TITLE
feat: add triage_review_comments tool for actionable threads only (#96)

### LINKED ISSUE #96: feat: add triage view — show only threads needing agent action
## Problem

Every review pass, the agent must:
1. Fetch all unresolved threads via `list_stack_review_comments`
2. Filter out PR-level review summaries (`is_pr_review`)
3. Filter to threads with no reply from the repo owner
4. Parse severity from emoji markers in comment bodies
5. Decide which threads need replies, code fixes, or GH issues

This filtering happens externally (often as inline Python parsing the raw JSON output) and is repeated on every single pass. It's the most common workflow in a review session.

## Proposed solution

Add a `triage_review_comments` tool (or a `triage=true` parameter on existing tools) that returns only threads needing agent action:

```
{
  "pr_number": 88,
  "needs_reply": [
    {"thread_id": "PRRT_...", "severity": "bug", "file": "comments.py", "line": 239, "title": "pagination missing"},
    {"thread_id": "PRRT_...", "severity": "info", "file": "stack.py", "line": 42, "title": "semantic change"}
  ],
  "needs_issue": [
    {"thread_id": "PRRT_...", "our_reply_mentions": "noted for followup", "has_issue_ref": false}
  ]
}
```

### Key filters
- **Exclude PR-level reviews** (`is_pr_review=true`) — these are just summary wrappers
- **Exclude threads with an owner reply** — already handled
- **Pre-classify severity** using reviewer adapters (already have this logic in `_classify_severity`)
- **Extract title** from the first bold text in the comment (Devin always uses `**title**` format)
- **Flag 'noted for followup' replies** that don't reference a GH issue number

### Benefits
- Eliminates repeated parsing on every review pass
- Makes the agent's first action deterministic: call triage → get action list → execute
- Reduces token usage (no need to return full comment bodies just to filter)

### LINKED ISSUE #106: Followup detection does not correlate across comments in a thread
## Problem

`_has_followup_without_issue` checks each comment independently via `any()`. If an owner posts comment A saying "Noted for followup" (no issue ref) and then comment B saying "Created #456 for this", the function still returns `True` because comment A alone satisfies all three conditions.

This means a thread where the owner already created an issue in a follow-up reply will still be flagged as `action="create_issue"`, potentially causing duplicate issue creation.

## Proposed Fix

Check if _any_ comment in the thread contains an issue reference before flagging. Something like:

```python
has_followup = any(c.author in owner_logins and _FOLLOWUP_KEYWORDS.search(c.body) for c in thread.comments)
has_issue_ref = any(_ISSUE_REF_RE.search(c.body) for c in thread.comments if c.author in owner_logins)
return has_followup and not has_issue_ref
```

## Context

Caught by Devin review on PR #104.

### PR DESCRIPTION
**Pull request by ichoosetoaccept**
## Summary

Add a new `triage_review_comments` MCP tool that returns only threads needing agent action — no noise, no full comment bodies. This is the most-requested workflow improvement: agents no longer need to fetch all threads and manually filter.

Also fixes the followup detection bug where issue refs in separate replies were not correlated across the thread.

## Changes

- **`models.py`**: New `TriageItem` and `TriageResult` models with severity, suggested action, snippet, and title extraction.
- **`tools/comments.py`**: New triage logic:
  - Filters to unresolved inline threads only (excludes PR-level reviews)
  - Excludes threads with an owner reply (`ichoosetoaccept` by default, configurable)
  - Pre-classifies severity using reviewer adapters (🔴 bug → `fix`, 📝 info → `reply`)
  - Extracts title from first bold text (`**Bug: title**` format)
  - Flags "noted for followup" replies missing a GH issue reference → `create_issue` action
  - `_has_followup_without_issue` checks for issue refs across all owner comments in the thread, not per-comment
  - Results sorted by severity (bugs first)
- **`server.py`**: Register tool, update workflow instructions to recommend triage as step 2.
- **`tests/test_triage.py`**: 16 tests covering all filtering, sorting, and edge cases.
- **`tests/test_mcp_integration.py`**: Updated expected tool count (9→10).

## Actions returned per thread

| Severity | Action | Meaning |
|----------|--------|---------|
| 🔴 bug, 🚩 flagged | `fix` | Needs a code change + regression test |
| 🟡 warning, 📝 info | `reply` | Acknowledge with `reply_to_comment` |
| Any (with followup) | `create_issue` | Owner said "noted for followup" but forgot `#N` |

Fixes #96
Closes #106

### CODE REVIEW

File: src/codereviewbuddy/tools/comments.py [hunk 1]
```diff
@@ -811,3 +821,144 @@ def _reply_to_pr_comment(
...
+
+
+def _has_owner_reply(thread: ReviewThread, owner_logins: frozenset[str]) -> bool:
+    """Check if any comment in the thread is from the repo owner / agent."""
+    return any(c.author in owner_logins for c in thread.comments)
+
+
+def _has_followup_without_issue(thread: ReviewThread, owner_logins: frozenset[str]) -> bool:
+    """Check if the owner replied with a 'noted for followup' but no issue reference."""
+    return any(c.author in owner_logins and _FOLLOWUP_KEYWORDS.search(c.body) and not _ISSUE_REF_RE.search(c.body) for c in thread.comments)
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply src/codereviewbuddy/tools/comments.py]: Valid concern. In practice, owners typically include the issue ref in the same comment as the followup note (that's the workflow we encourage). But the cross-comment correlation case is real. Tracked in #106.

File: tests/test_triage.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+class TestTriageReviewComments:
+    """Integration tests that mock list_review_comments and verify triage logic."""
+
+    def _mock_list(self, mocker: MockerFixture, threads: list[ReviewThread]) -> AsyncMock:
+        return mocker.patch(
+            "codereviewbuddy.tools.comments.list_review_comments",
+            new_callable=AsyncMock,
+            return_value=ReviewSummary(threads=threads),
+        )
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply tests/test_triage.py]: Intentional — unit tests mock `list_review_comments` to isolate the triage filtering/classification logic. The `status="unresolved"` filter passthrough is covered by the integration path in `test_mcp_integration.py` (tool registration + schema). A full end-to-end test mocking the GitHub API would be nice but adds complexity for marginal coverage gain.

File: src/codereviewbuddy/tools/comments.py [hunk 3]
```diff
@@ -811,3 +821,143 @@ def _reply_to_pr_comment(
...
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply src/codereviewbuddy/tools/comments.py]: Good catch — fixed. `_has_followup_without_issue` now checks for followup keywords and issue refs across all owner comments in the thread, not per-comment. Added regression test `test_issue_ref_in_separate_reply`.

File: src/codereviewbuddy/tools/comments.py [hunk 4]
```diff
@@ -811,3 +821,143 @@ def _reply_to_pr_comment(
...
+        repo: Repository in "owner/repo" format. Auto-detected if not provided.
+        owner_logins: GitHub usernames considered "ours" (agent + human).
+            Defaults to ``["ichoosetoaccept"]`` if not provided.
+        cwd: Working directory for git operations.
+        ctx: FastMCP context for progress reporting.
+
+    Returns:
+        TriageResult with only actionable items, sorted by severity.
+    """
+    from codereviewbuddy.tools.stack import _classify_severity
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply src/codereviewbuddy/tools/comments.py]: Acknowledged. The deferred import avoids circular deps and the function is stable. If it moves, tests will catch it immediately. Not worth duplicating or extracting to a shared module for a 2-line function.

File: src/codereviewbuddy/tools/comments.py [hunk 5]
```diff
@@ -811,3 +821,148 @@ def _reply_to_pr_comment(
...
+
+# ---------------------------------------------------------------------------
+# Triage — actionable threads only (#96)
+# ---------------------------------------------------------------------------
+
+_BOLD_TITLE_RE = re.compile(r"\*\*(?:Bug|Info|Warning|Flagged)?:?\s*(.+?)\*\*", re.IGNORECASE)
+_ISSUE_REF_RE = re.compile(r"#\d+")
+_FOLLOWUP_KEYWORDS = re.compile(r"noted for followup|tracked for later|will address later|followup", re.IGNORECASE)
+
+_SEVERITY_ORDER = {"bug": 0, "flagged": 1, "warning": 2, "info": 3}
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply src/codereviewbuddy/tools/comments.py]: Confirmed — `Severity.value` returns lowercase strings (`"bug"`, `"info"`, etc.) which match the keys in `_SEVERITY_ORDER` and the string comparisons in `_classify_action`.

**Comment by ichoosetoaccept** [comment]
### Merge activity

* **Feb 11, 2:25 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/detailobsessed/codereviewbuddy/104).
* **Feb 11, 2:27 PM UTC**: [Graphite](https://app.graphite.com/github/pr/detailobsessed/codereviewbuddy/104) rebased this pull request as part of a merge.
* **Feb 11, 2:27 PM UTC**: @ichoosetoaccept merged this pull request with [Graphite](https://app.graphite.com/github/pr/detailobsessed/codereviewbuddy/104).
